src/main/java/org/opentripplanner/middleware/controllers/api/MonitoredTripController.java

```diff
@@ -56,4 +107,30 @@ private void verifyBelowMaxNumTrips(String userId, Request request) {
             );
         }
     }
+
+    /**
+     * Checks that non-realtime itineraries exist for the days the specified monitored trip is active.
+     */
+    private static ItineraryExistenceChecker.Result checkItineraryExistence(MonitoredTrip trip, Request request) {
+        ItineraryExistenceChecker itineraryChecker = new ItineraryExistenceChecker(OtpDispatcher::sendOtpPlanRequest);
+        try {
+            ItineraryExistenceChecker.Result checkResult = itineraryChecker.checkAll(ItineraryUtils.getItineraryExistenceQueries(trip));
+            if (!checkResult.allItinerariesExist) {
+                logMessageAndHalt(
+                    request,
+                    HttpStatus.BAD_REQUEST_400,
+                    "An itinerary does not exist for some of the monitored days for the requested trip."
+                );
+            }
```

Comment:
In order for the UI to be able to show which days of the week the trip is possible, shouldn't this return some kind of map of the days of the week that the trip is possible?

Reply:
This method is called on PUT or POST. But the next step is indeed to create an endpoint for the UI to report the checks... Should that be in this PR or a separate one?

Reply:
I guess I'll go ahead and build the endpoint for the UI too.